internal/server/data/provideruser_test.go

```diff
@@ -73,17 +73,13 @@ func TestSyncProviderUser(t *testing.T) {
 					err = CreateIdentity(db, user)
 					assert.NilError(t, err)
 
-					pu := &models.ProviderUser{
-						ProviderID: provider.ID,
-						IdentityID: user.ID,
+					pu, err := CreateProviderUser(db, provider, user)
```

Comment:
I'm a little confused by this change. Was it not necessary to create the provider user explicitly before? Why is it needed now?

Reply:
I suspect because we were previously using `gorm.DB.Save` that the update below was actually using sql `INSERT` once it found there was not existing record.

Now that we explicitly only use sql `UPDATE` in update functions we have to insert it first.

We might be able to insert all the values in `CreateProviderUser`, instead of calling update immediately after.

Reply:
Yeah, the update was used as a a "create or update" function before. Which our new insert doesn't do. Creating a provider user doesn't allow specifying these fields, so we have to use the update to add them. We might need a logic change here in a different PR